src/Evolution/Systems/NewtonianEuler/ComputeItems.cpp

```diff
@@ -100,6 +140,7 @@ Scalar<DataType> specific_kinetic_energy(
 
 #define DTYPE(data) BOOST_PP_TUPLE_ELEM(0, data)
 #define DIM(data) BOOST_PP_TUPLE_ELEM(1, data)
+#define THERMO_DIM(data) BOOST_PP_TUPLE_ELEM(1, data)
```

Comment:
I think what you've done here (setting DIM, THERMO_DIM both to the the 1'th element, since you know they won't be used together) is correct, but I found I needed to read the whole instantiation section a few times to make sure I understood it.

I would favor that the types be defined and undef'd as needed around each instantiation block. It's a few more lines of code, but avoids some possible ambiguities.